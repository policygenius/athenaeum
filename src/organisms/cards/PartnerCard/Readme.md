### PartnerCard Example:

```jsx
    <PartnerCard>
      <strong>New Component Created!</strong><br /><small>Now go and write some code! -- <em>You code monkey, you ;)</em></small>.
    </PartnerCard>
```
