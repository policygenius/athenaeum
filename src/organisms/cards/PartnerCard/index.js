import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Icon from 'atoms/Icon';
import Button from 'atoms/Button';
import styles from './partner_card.module.scss';

function PartnerCard(props) {
  const {
    className,
    icon,
    actionCopy,
    onButtonClick,
    actionLink,
    centerLabel,
    centerText
  } = props;

  const classes = [
    styles['partner-card'],
    className,
  ];

  return (
    <div className={classnames(...classes)}>
      <div className={styles['cell']}>
        <Icon icon={icon} />
      </div>
      <div className={styles['cell']}>
        <h2 className={styles['label']}>{centerLabel}</h2>
        <span className={styles['a5']}>
          {centerText}
        </span>
      </div>
      <div className={styles['cell']}>
        <Button
          variant='info'
          outline
          text={actionCopy}
          href={actionLink}
          onClick={onButtonClick}
        />
      </div>
    </div>
  );
}


PartnerCard.propTypes = {
  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,
  /**
   * The name of the icon to render
   */
  icon: PropTypes.string,
  /**
   * Text inside the 'Call to Action' button
   */
  actionCopy: PropTypes.string,
  /**
   * 'Call to Action' button link
   */
  actionLink: PropTypes.string,
  /**
   * Alternatively, may provide a callback for the button
   */
  onButtonClick: PropTypes.func,
  /**
   * Provides context for the center text
   */
  centerLabel: PropTypes.string,
  /**
   * Short and snappy center text
   */
  centerText: PropTypes.string.isRequired
};

PartnerCard.defaultProps = {
  actionCopy: 'Shop Now',
  centerLabel: 'BBB Rating',
  centerText: 'A',
  icon: 'pencil'
};

export default PartnerCard;
