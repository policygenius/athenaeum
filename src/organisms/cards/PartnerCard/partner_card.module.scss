@import '../shared/cards.module.scss';

.partner-card {
  width: 100%;
  border: 1px solid color('neutral-4');
}

.cell {
  padding: ru(.75);
  text-align: center;
}

.label {
  @extend %label;
}

.a5 {
  @include typography-a-5-bold();
}

@media #{$tablet} {
  .partner-card {
    display: table;
    table-layout: fixed;
    border: 1px solid color('neutral-4');
  }

  .cell {
    display: table-cell;
    padding: ru(1.5) ru(.75);
    vertical-align: middle;
  }

  .cell:first-of-type {
    padding-left: ru(1.5);
    text-align: left;
  }

  .cell:last-of-type {
    padding-right: ru(1.5);
  }
}
