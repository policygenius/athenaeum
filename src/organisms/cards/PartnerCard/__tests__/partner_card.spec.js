import React from 'react';
// import sinon from 'sinon';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import PartnerCard from 'organisms/cards/PartnerCard';

describe('<PartnerCard />', () => {
  // let defaultProps;
  //
  // beforeEach(() => {
  //   defaultProps = {
  //     // Add default props here
  //   };
  // });

  it('renders correctly', () => {
    const wrapper = mount(<PartnerCard />);

    expect(wrapper.type()).toEqual(PartnerCard);
  });

  it('renders correctly', () => {
    const actual = renderer.create(<PartnerCard />).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
