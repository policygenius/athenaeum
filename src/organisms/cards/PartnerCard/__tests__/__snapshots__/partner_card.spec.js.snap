// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PartnerCard /> renders correctly 1`] = `
<div
  className="partner-card"
>
  <div
    className="cell"
  >
    <span
      className="icon-wrapper"
      style={
        Object {
          "height": undefined,
          "width": undefined,
        }
      }
    >
      <img
        alt="pencil"
        className="img"
        src="https://static.policygenius.com/svgs/pencil.svgz"
        title="pencil"
      />
    </span>
  </div>
  <div
    className="cell"
  >
    <h2
      className="label"
    >
      BBB Rating
    </h2>
    <span
      className="a5"
    >
      A
    </span>
  </div>
  <div
    className="cell"
  >
    <button
      className="button info outline"
      type="button"
    >
      Shop Now
    </button>
  </div>
</div>
`;
